### ISSUE TITLE
linter: Oxlint should not discourage explicit list semantics in `no-redundant-roles`

### ISSUE THREAD

**Issue by mariusschulz** [oxc-project/oxc#22792]
### What version of Oxlint are you using?

1.67.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```json
{
  "plugins": [
    "jsx-a11y"
  ],
  "categories": {
    "correctness": "error"
  },
  "rules": {}
}
```

### What happened?

Oxlint should not discourage setting an explicit `role="list"` on a `<ul>` or `<ol>`. When `list-style: none` is set, [Safari removes list semantics](https://gerardkcohen.me/writing/2017/voiceover-list-style-type.html) which can be undesired from an accessibility perspective. 

```html
<style>
  ul, ol {
    list-style: none;
  }
</style>
```

```tsx
function Component() {
  return (
    <ul role="list">
      <li role="listitem">Test</li>
    </ul>
  );
}
```

The `no-redundant-roles` rule started highlighting this in Oxlint v1.64.0 (see https://github.com/oxc-project/oxc/pull/22069):

```
× [jsx-a11y(no-redundant-roles)](https://oxc.rs/docs/guide/usage/linter/rules/jsx_a11y/no-redundant-roles.html): The `ul` element has an implicit role of `list`. Defining this explicitly is redundant and should be avoided.
   ╭─[index.tsx:3:9]
 2 │   return (
 3 │     <ul role="list">
   ·         ───────────
 4 │       <li role="listitem">Test</li>
   ╰────
  help: Remove the redundant role `list` from the element `ul`.

  × [jsx-a11y(no-redundant-roles)](https://oxc.rs/docs/guide/usage/linter/rules/jsx_a11y/no-redundant-roles.html): The `li` element has an implicit role of `listitem`. Defining this explicitly is redundant and should be avoided.
   ╭─[index.tsx:4:11]
 3 │     <ul role="list">
 4 │       <li role="listitem">Test</li>
   ·           ───────────────
 5 │     </ul>
   ╰────
  help: Remove the redundant role `listitem` from the element `li`.
```

I would recommend that we special-case list roles in the `no-redundant-roles` rule so that the rule won't produce lint warnings/errors for these explicit accessibility fixes.

**Comment by stefanprobst** [comment]
x-ref https://github.com/oxc-project/oxc/issues/18864
